src/main/java/harmonised/pmmo/config/codecs/AttributeModifier.java

```diff
@@ -0,0 +1,19 @@
+package harmonised.pmmo.config.codecs;
+
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+
+public record AttributeModifier(
+	String attribute,
+	double amount,
+	String operation
```

Comment:
use the `AttrbiuteModifier.Operation` class instead of String.

Reply:
It seems that AttributeModifier.Operation does not have a codec. I see that it's an enum, but it cannot use the com.mojang.serialization.Codec.INT codec. Any suggestions?

Reply:
```java
public static final Codec<AttributeModifier.Operation> OPERATION_CODEC = ExtraCodecs.stringResolverCodec(AttributeModifier.Operation::name,
	str -> switch (str) {
		case "MULTIPLY_BASE" -> AttributeModifier.Operation.MULTIPLY_BASE;
		case "MULTIPLY_TOTAL" -> AttributeModifier.Operation.MULTIPLY_TOTAL;
		default -> AttributeModifier.Operation.ADDITION;
});
```
i'd put this in `CodecTypes` under the `DAMAGE_XP_CODEC`

Another option is to use integers, but i feel like that's less readable.  If you did go that route though, you could do 
```java
public static final Codec<AttributeModifier.Operation> OPERATION_CODEC = Codec.INT.xmap(AttributeModifier.Operation::fromValue, AttributeModifier.Operation::toValue);
```

Reply:
The first option looks way better and it's easier to document for the users, we'll go with that.